external/mmdetection/detection_tasks/apis/detection/ote_utils.py

```diff
@@ -96,6 +101,57 @@ def get_task_class(path):
     return getattr(module, class_name)
 
 
+def get_actmap(features, output_res):
+    am = cv2.resize(features, output_res, interpolation=cv2.INTER_NEAREST)
+    denominator = (np.max(am) - np.min(am) + 1e-12) / 255
+    am = (am - np.min(am)) / denominator
+    am = np.uint8(np.floor(am))
+    return am
+
+
+def draw_instance_segm_saliency_map(predictions, dataset_item, labels):
+    """
+    Converts predictions to masks and accumulate them to saliency map (naive way).
+    """
+    width, height = dataset_item.width, dataset_item.height
+    aggregated_mask = np.zeros([len(labels), height, width], dtype=np.uint8)
+    for prediction in predictions:
+        if not isinstance(prediction.shape, Polygon):
+            continue
+        contours = np.array(
+            [[(int(p.x * width), int(p.y * height)) for p in prediction.shape.points]]
+        )
+        assert len(prediction.get_labels()) == 1
+        scored_label = prediction.get_labels()[0]
+        label_idx = labels.index(scored_label.label)
+        color = int(scored_label.probability * 255)
+        cv2.drawContours(aggregated_mask[label_idx], contours, -1, color, -1)
+    return aggregated_mask
+
+
+def add_feature_info_to_data_item(feature_info, dataset_item, model, labels, return_saliency_map):
+    """ Assign feature (representation) vector and saliency map to predictions dataset_item. """
+
+    feature_vector, feature_map = feature_info
+    active_score = TensorEntity(name="representation_vector", numpy=feature_vector.reshape(-1))
```

Comment:
name it as `feature_vec(tor)` or `representation_vec(tor)`. Active score is another kind of metadata. You can have a look at classification task implementation.

Reply:
Good point, fixed